### PR TITLE
[server/analytics] fix filter for unconfirmed user

### PR DESCRIPTION
**Pull request by AlexTugarev**
Resolves remaining issue with https://github.com/gitpod-io/gitpod/issues/4296

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4944#" title="Author self-approved">AlexTugarev</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/auth/generic-auth-provider.ts [hunk 1]
```diff
@@ -357,7 +357,7 @@ export class GenericAuthProvider implements AuthProvider {
                 message = 'OAuth Error. Please try again.'; // this is a 5xx response from authorization service
             }
 
-            if (!UnconfirmedUserException.is(error)) {
+            if (!UnconfirmedUserException.is(err)) {
```
Comment (AlexTugarev) [inline components/server/src/auth/generic-auth-provider.ts:360]: Missed that during review.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 7e1f13306b4c1b7df6385d30d1c23505bb18dda2</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4944#" title="Author self-approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4944#issuecomment-886416904" title="LGTM">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**